File: ios-native/src/types/Mission.ts
export interface FlightLeg {
  id: string;
  legNumber: number;
  departure: string;
  arrival: string;
  departureTime: string;
  arrivalTime: string;
  flightHours: number;
  distanceNm?: number;
  tailNumber?: string;
  remarks?: string;
  pax?: number;
  cargo?: number;
}

export interface Mission {
  id: number;
  missionNumber: string;
  missionType: string;
  status: "In Progress" | "Completed" | "Deployment" | "Planning";
  dateRange: string;
  legs: number;
  pax: number;
  cargo: number;
  aircraftType?: string;
  createdAt: string;
  photos?: string[];
  notes?: string;
  flightLegs?: FlightLeg[];
  totalFlightHours?: number;
  crewInfo?: CrewInfo;
}

export interface FCCMember {
  id: string;
  name: string;
  rank: string;
  employeeNumber: string;
  sdap: boolean;
  megp: boolean;
  departureDate: string;
  returnDate: string;
  distinguishedVisitors: string;
}

export interface MaintenancePersonnel {
  id: string;
  name: string;
  rank: string;
  employeeNumber: string;
  sdap: boolean;
  megp: boolean;
  departureDate: string;
  returnDate: string;
  fccMonitor: string;
}

export interface AircraftCommander {
  name: string;
  rank: string;
  flyingSquadron: string;
}

export interface CrewInfo {
  fccMembers: FCCMember[];
  maintenancePersonnel: MaintenancePersonnel[];
  aircraftCommander: AircraftCommander;
}

export interface Airport {
  icao: string;
  name: string;
  city: string;
  country: string;
  latitude: number;
  longitude: number;
  similarity?: number;
}

export interface FlightStatistics {
  totalFlightHours: number;
  totalDistanceNm: number;
  totalCargoMoved: number;
  totalPaxMoved: number;
  missionsCompleted: number;
  averageFlightHours: number;
  longestFlight: number;
  mostFrequentRoute: string;
}

File: swift-ios-app/AMCMissionTracker/Models/Mission.swift
import Foundation

struct Mission: Identifiable, Codable {
    let id = UUID()
    var missionNumber: String
    var missionType: String
    var status: MissionStatus
    var startDate: Date
    var endDate: Date
    var legs: [FlightLeg]
    var pax: Int
    var cargo: Double
    var aircraftType: String
    var notes: String
    var photos: [String]
    var crewInfo: CrewInfo?
    
    var totalFlightHours: Double {
        legs.reduce(0) { $0 + $1.flightHours }
    }
    
    var totalDistance: Double {
        legs.reduce(0) { $0 + $1.distanceNm }
    }
    
    var dateRange: String {
        let formatter = DateFormatter()
        formatter.dateStyle = .short
        return "\(formatter.string(from: startDate)) - \(formatter.string(from: endDate))"
    }
    
    init(missionNumber: String = "", 
         missionType: String = "Transport",
         status: MissionStatus = .planning,
         startDate: Date = Date(),
         endDate: Date = Calendar.current.date(byAdding: .day, value: 7, to: Date()) ?? Date(),
         legs: [FlightLeg] = [],
         pax: Int = 0,
         cargo: Double = 0.0,
         aircraftType: String = "C-17A",
         notes: String = "",
         photos: [String] = [],
         crewInfo: CrewInfo? = nil) {
        self.missionNumber = missionNumber
        self.missionType = missionType
        self.status = status
        self.startDate = startDate
        self.endDate = endDate
        self.legs = legs
        self.pax = pax
        self.cargo = cargo
        self.aircraftType = aircraftType
        self.notes = notes
        self.photos = photos
        self.crewInfo = crewInfo
    }
}

enum MissionStatus: String, CaseIterable, Codable {
    case planning = "Planning"
    case inProgress = "In Progress"
    case completed = "Completed"
    case deployment = "Deployment"
    
    var color: String {
        switch self {
        case .planning: return "yellow"
        case .inProgress: return "blue"
        case .completed: return "green"
        case .deployment: return "red"
        }
    }
}